security-command-center/snippets/src/test/java/management/api/EventThreatDetectionCustomModuleTest.java

```diff
@@ -0,0 +1,166 @@
+import com.google.cloud.securitycentermanagement.v1.ListEventThreatDetectionCustomModulesRequest;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient.ListEventThreatDetectionCustomModulesPagedResponse;
+import com.google.cloud.testing.junit4.MultipleAttemptsRule;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.StreamSupport;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class EventThreatDetectionCustomModuleTest {
+  // TODO(Developer): Replace the below variable
+  private static final String PROJECT_ID = System.getenv("SCC_PROJECT_ID");
+  private static final String CUSTOM_MODULE_DISPLAY_NAME = "java_sample_custom_module_test";
+  private static final int MAX_ATTEMPT_COUNT = 3;
+  private static final int INITIAL_BACKOFF_MILLIS = 120000; // 2 minutes
+
+  @Rule
+  public final MultipleAttemptsRule multipleAttemptsRule =
+      new MultipleAttemptsRule(MAX_ATTEMPT_COUNT, INITIAL_BACKOFF_MILLIS);
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("SCC_PROJECT_ID");
+  }
+
+  @AfterClass
+  public static void cleanUp() throws IOException {
+    // Perform cleanup after running tests
+    cleanupExistingCustomModules();
+  }
+
+  // cleanupExistingCustomModules clean up all the existing custom module
+  private static void cleanupExistingCustomModules() throws IOException {
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+      ListEventThreatDetectionCustomModulesRequest request =
+          ListEventThreatDetectionCustomModulesRequest.newBuilder()
+              .setParent(String.format("projects/%s/locations/global", PROJECT_ID))
+              .build();
+      ListEventThreatDetectionCustomModulesPagedResponse response =
+          client.listEventThreatDetectionCustomModules(request);
+      // Iterate over the response and delete custom module one by one which start with
+      // java_sample_custom_module
+      for (EventThreatDetectionCustomModule module : response.iterateAll()) {
+        if (module.getDisplayName().startsWith("java_sample_custom_module")) {
+          String customModuleId = extractCustomModuleId(module.getName());
+          deleteCustomModule(PROJECT_ID, customModuleId);
+        }
+      }
+    }
```

Comment:
nit: consider silencing exception. it is better to leave leftovers with and log the problem than fail a lot of tests.